src/sipmpdf/functions.py

```diff
@@ -218,31 +219,144 @@ def darkcurrent_response(x: np.float64,
 
+  x=extend_arr(x)
+  smear=extend_arr(smear)
+  gain=extend_arr(gain)
+  resolution=extend_arr(resolution)
+  
+  delta = 8 * smear / n_samples  # Getting the integration window
+  idx = kern.local_index(x, axis=0)
+  x_pr = -4 * smear + delta * (idx + 0.5)
+
+  # Given that the computed result will exhibit periodicity within the
+  # integration window, here we compute a simple rolling means with 5 parts
+  return kern.sum(darkcurrent_response(x - x_pr, gain, resolution)*normal(x_pr, 0, smear)*delta,axis=0)
+
+@expand_shape
+def afterpulsing_summation(x: np.float64,
+                           total: np.int64,
+                           ap_smear: np.float64, ##which should this actually be?
+                           ap_beta: np.float64, 
+                           ap_prob: np.float64,
+                           pedestal: np.float64,
+                           gain: np.float64) ->np.float64:
+  kern = kernel_switch(x, ap_smear, ap_prob, ap_beta, total,pedestal,gain)
+  k=10
+  ##come back to upgrade this later, decide on most effective method to ensure accuracy and not use too much computing power
+  def extend_arr(x):
+    return kern.repeat(x[kern.newaxis,...], k, axis=0)
+    
+  x=extend_arr(x)
+  ap_smear=extend_arr(ap_smear)
+  ap_beta=extend_arr(ap_beta)
+  ap_prob=extend_arr(ap_prob)
+  total=extend_arr(total)
+  pedestal=extend_arr(pedestal)
+  gain=extend_arr(gain)
+  
+  idx = kern.local_index(x, axis=0)
+  idx+=1
+
+  return kern.sum(binomial_prob(x=idx,total=total,prob=ap_prob)*ap_response_smeared(x=x-(pedestal+total*gain),smear=ap_smear,n_ap=idx,beta=ap_beta), axis=0)
+
+def k_summation(
+    x: np.float64,
+    common_noise: np.float64,
+    pixel_noise: np.float64,
+    ap_beta: np.float64,
+    ap_prob: np.float64,
+    pedestal: np.float64,
+    gain: np.float64,
+    poisson_mean: np.float64,
+    poisson_borel: np.float64)->np.float64:
+  kern = kernel_switch(x, common_noise, pixel_noise, ap_prob, ap_beta, pedestal, gain, poisson_mean, poisson_borel)
+  
+  k_max=10
+  ##come back to upgrade this later, decide on most effective method to ensure accuracy and not use too much computing power
+  def extend_arr(x):
+    return kern.repeat(x[kern.newaxis,...], k_max, axis=0)
+    
+  x=extend_arr(x)
+  common_noise=extend_arr(common_noise)
+  pixel_noise=extend_arr(pixel_noise)
+  ap_beta=extend_arr(ap_beta)
+  ap_prob=extend_arr(ap_prob)
+  pedestal=extend_arr(pedestal)
+  gain=extend_arr(gain)
+  poisson_mean=extend_arr(poisson_mean)
+  poisson_borel=extend_arr(poisson_borel)
+  
+  idk = kern.local_index(x, axis=0)
+  idk+=1
+  sigma_k = kern.sqrt(common_noise**2 + idk*pixel_noise**2)
+
+  return kern.sum(generalized_poisson(k=idk,mean=poisson_mean,borel=poisson_borel)*((binomial_prob(x=0, total=idk, prob=ap_prob)*normal(x=x, mean=pedestal+idk*gain, scale=sigma_k))+afterpulsing_summation(x=x,total=idk,ap_smear=common_noise,ap_beta=ap_beta,ap_prob=ap_prob,gain=gain,pedestal=pedestal)),axis=0)
+  
```

Comment:
- Can you add a function of `sipm_response_no_dark` which is effectively, the `sipm_response` function with `dc_prob` fixed at 0? 

- Also, help rename `normal_smeared_poisson` to `sipm_response_no_dark_no_ap`.